src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.hpp

```diff
@@ -242,6 +258,24 @@ class AbstractASTNode : public std::enable_shared_from_this<AbstractASTNode> {
   // it, the reference cannot be resolved (see knows_table) and std::nullopt is returned.
   std::optional<NamedColumnReference> _resolve_local_alias(const NamedColumnReference& named_column_reference) const;
 
+  // A helper method that can be used by all subclasses to do most of the work needed for clone_subtree. This sets the
+  // parent<->children relation - all that's left to do in the subclasses is to deep-copy all members that are not
+  // deep copied by default (i.e., you will have to copy shared_ptrs by hand, but strings not)
+  template <typename Subclass>
+  std::shared_ptr<std::decay_t<Subclass>> _clone_without_subclass_members() const {
```

Comment:
don't like the "without_subclass_members" parts, because you are copying subclass members by invoking the copy constructor? 
Also, why use `std::decay_t` here? It's nice to make people look up what the hell that is, but I'd rather have this function invoked in a way that doesn't require such esoterics.

Reply:
Here, `decay_t<T>` is essentially `std::remove_reference_t<std::remove_const_t<T>>`. Replacing all uses of decay with that bloats the code and I haven't found a neat way to use `using` here. Would you consider the long version better?